### PR TITLE
Add documentation for session properties manager

### PR DESCRIPTION
**Pull request by rongrong**

### CODE REVIEW

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
+==========================
+Session Properties Manager
+==========================
+
+Session properties manager provides a way to dynamically override default
```
Comment (raghavsethi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: A session properties manager plugin

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+session properties. It supports overriding default session properties for each
+session based on ``SessionConfigurationContext``, which includes ``user``,
+``source`` and ``clientTags``. The returned properties override the default
+values but not the final values, thus any user-provided values will override
```
Comment (raghavsethi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: The properties returned override the default values specified in code. However, session properties set by users are always respected, and take precedence over these values.

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+the values returned by session properties manager.
+
+Implementation
+--------------
+
+``SessionPropertyConfigurationManagerFactory`` is responsible for creating an
+``SessionPropertyConfigurationManager`` instance. Implementations of
+``SessionPropertyConfigurationManager`` should include two methods:
+
+* ``getSystemSessionProperties``: Return a map of system session properties.
```
Comment (raghavsethi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: Returns

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+* ``getCatalogSessionProperties``: Return a map of catalog to catalog properties map.
```
Comment (raghavsethi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: Returns a map of catalog to catalog properties.

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+The implementation of ``SessionPropertyConfigurationManager`` and
```
Comment (raghavsethi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: This paragraph is redundant

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 6]
```diff
@@ -0,0 +1,25 @@
+==========================
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: should we rename this file to `session-properties-manager.rst`?

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+Session Properties Manager
+==========================
+
+Session properties manager provides a way to dynamically override default
+session properties. It supports overriding default session properties for each
+session based on ``SessionConfigurationContext``, which includes ``user``,
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: `based on the`

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+``source`` and ``clientTags``. The returned properties override the default
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: ``source``, and ``clientTags` (`,` after the second `and`)

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+values but not the final values, thus any user-provided values will override
+the values returned by session properties manager.
+
+Implementation
+--------------
+
+``SessionPropertyConfigurationManagerFactory`` is responsible for creating an
+``SessionPropertyConfigurationManager`` instance. Implementations of
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: `creating an instance of`

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+``SessionPropertyConfigurationManager`` should include two methods:
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: `include` -> `provide`

File: presto-docs/src/main/sphinx/admin/session-properties.rst (hunk 4 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/admin/session-properties.rst]: `catalog to catalog properties map` sounds confusing.

File: presto-docs/src/main/sphinx/admin/session-properties.rst [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+session properties. It supports overriding default session properties for each
+session based on ``SessionConfigurationContext``, which includes ``user``,
+``source`` and ``clientTags``. The returned properties override the default
+values but not the final values, thus any user-provided values will override
+the values returned by session properties manager.
+
+Implementation
+--------------
+
+``SessionPropertyConfigurationManagerFactory`` is responsible for creating an
```
Comment (nezihyigitbasi) [inline presto-docs/src/main/sphinx/admin/session-properties.rst]: The ``SessionPropertyConfigurationManagerFactory``

File: presto-docs/src/main/sphinx/admin/session-properties.rst (hunk 2 above)
Comment (rongrong) [reply presto-docs/src/main/sphinx/admin/session-properties.rst]: The original text was the same as the java doc in code, which was reviewed by @electrum. I do think your wording is clearer though.

File: presto-docs/src/main/sphinx/admin/session-properties.rst (hunk 4 above)
Comment (rongrong) [reply presto-docs/src/main/sphinx/admin/session-properties.rst]: I agree. Not sure what's a better way to describe it without a long paragraph...

**Comment by rongrong** [comment]
@nezihyigitbasi  @raghavsethi  Any more comments on this?

File: presto-docs/src/main/sphinx/admin/session-properties-manager.rst [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+==========================
+
+A session properties manager plugin provides a way to dynamically override default
+session properties. It supports overriding default session properties for each
+session based on the ``SessionConfigurationContext``, which includes ``user``,
+``source``, and ``clientTags``. The returned properties override the default
+values. However, session properties set by users are always respected, and take
+precedence over these values.
+
+Implementation
```
Comment (electrum) [inline presto-docs/src/main/sphinx/admin/session-properties-manager.rst]: I don’t think this is useful for documentation. It basically just repeats the Javadoc and is the type of thing that will become out of date as the code changes.

Comment (rongrong) [reply presto-docs/src/main/sphinx/admin/session-properties-manager.rst]: In that case I don't really see it necessary to have documentation for this feature. We don't have a open source plugin implemented, so there's not much of "how to configure it" to document.

**Comment by electrum** [comment]
We still need to tell someone how to configure it if they write a plugin.

**Comment by raghavsethi** [comment]
@rongrong could you and @electrum figure out what's required here and merge/close?

**Comment by rongrong** [comment]
@electrum Do we still want this? Is there an example of documenting how to configure a plugin?

**Comment by raghavsethi** [comment]
Given @rongrong 's comment (https://github.com/prestodb/presto/pull/9349#discussion_r153379739) and my documentation for the new built-in implementation https://github.com/prestodb/presto/commit/d586e2f377165b583a051cf053168bffed500233 it looks like we no longer need this?

**Pull request closed without merging**